scripts/tests/yaml/runner.py

```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/env -S python3 -B
+
+# Copyright (c) 2023 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from relative_importer import *  # isort: split
+
+import sys
+import traceback
+
+import click
+from matter_yamltests.definitions import SpecDefinitions, SpecDefinitionsFromPaths
+from matter_yamltests.parser import TestParser, TestParserConfig
+from matter_yamltests.parser_builder import TestParserBuilderConfig, TestParserBuilderOptions
+from matter_yamltests.parser_config import TestConfigParser
+from matter_yamltests.pseudo_clusters.pseudo_clusters import PseudoClusters, get_default_pseudo_clusters
+from matter_yamltests.runner import TestRunner, TestRunnerConfig, TestRunnerOptions
+from matter_yamltests.websocket_runner import WebSocketRunner, WebSocketRunnerConfig
+from paths_finder import PathsFinder
+from tests_finder import TestsFinder
+from tests_logger import TestParserLogger, TestRunnerLogger, WebSocketRunnerLogger
+
+#
+# Options
+#
+
+
+def test_parser_options(f):
+    f = click.option("--configuration_name", type=str, required=True, show_default=True,
```

Comment:
That is a leftover. Removed.

Reply:
See https://github.com/project-chip/connectedhomeip/pull/25155#discussion_r1114261502 and https://github.com/project-chip/connectedhomeip/pull/25155#discussion_r1114267051

Reply:
What I am trying to say is that having `required=True` is doing nothing here since you are providing a default. `required=True` is supposed to mean that if you do not pass the command line argument and there is not value set to `configuration_name` that the command should fail, but you provide a default so this argument is not actually a required command line argument